sdks/java/core/src/main/java/org/apache/beam/sdk/io/WriteFiles.java

```diff
@@ -394,6 +418,7 @@ public void startBundle(StartBundleContext c) {
 
     @ProcessElement
     public void processElement(ProcessContext c, BoundedWindow window) throws Exception {
+      sink.getDynamicDestinations().setSideInputAccessorFromProcessContext(c);
```

Comment:
This is what we did in BigQueryIO because the pattern is that implementers of DynamicDestinations don't necessarily know it's being called from inside a processElement. We wanted to be free to call it from processTimer, or something else completely. Basically users don't interact directly with WriteFiles ever.